python_ta/checkers/redundant_assignment_checker.py

```diff
@@ -34,33 +34,47 @@ class RedundantAssignmentChecker(BaseChecker):
     name = "redundant_assignment"
     msgs = {
         "E9959": (
-            "This assignment statement is redundant;" " You can remove it from the program.",
+            "This assignment statement for variable(s) `%s` is redundant;"
+            " You can remove it from the program.",
             "redundant-assignment",
             "This assignment statement is redundant;" " You can remove it from the program.",
         )
     }
 
     def __init__(self, linter=None) -> None:
         super().__init__(linter=linter)
-        self._redundant_assignment: set[nodes.Assign] = set()
+        self._redundant_assignment: dict[str, nodes.Assign] = {}
```

Comment:
As you noted, your presentation last week, the data structure here is awkward because there are multiple times where you are looking up a node from the values of the dictionary.

A better option for this attribute is something like `dict[nodes.Assign, list[str]]`, where for each assignment node the `list[str]` is a list of the target names that are redundant (and therefore displayed in the error message).

Also, I suspect the correct key type is `nodes.Assign | nodes.AugAssign | node.AnnAssign` (not sure if they share a specific common superclass).